src/domains/chart/components/disable-out-of-view.tsx

```diff
@@ -65,12 +65,15 @@ export const DisableOutOfView = ({
   /* separate functionality - adding custom styles to portalNode */
   const chartSettings = chartLibrariesSettings[attributes.chartLibrary]
   const [hasPortalNodeBeenStyled, setHasPortalNodeBeenStyled] = useState<boolean>(false)
-  // todo omit this for Cloud
+  const isShowingAlarmOnChart = useSelector(selectAlarm)?.chart === attributes.id
   useLayoutEffect(() => {
     if (hasPortalNodeBeenStyled) {
       return
     }
-    const styles = getPortalNodeStyles(attributes, chartSettings)
+    const shouldAddSpecialHeight = isShowingAlarmOnChart
+      && attributes.chartLibrary === "dygraph"
+      && chartSettings.hasLegend(attributes)
+    const styles = getPortalNodeStyles(attributes, chartSettings, shouldAddSpecialHeight)
```

Comment:
Which part of the alarm adds special height to the chart? As I see the alarm component is inside the boundaries of the chart component.
I am asking because I should update the virtualized dashboard as well.

Reply:
The idea is that when `alarm_when` hash param is in the url, then the chart should have fixed height to around 40% of the screen. And this adjustment happens only on mount - so it has similar effect as having `chart_height` properties in localStorage on init.

Correct me if i'm wrong, but if settings in localStorage didn't need any handling in virtualised dashboard, this shouldn't be a problem too

Reply:
There is the `useChartEstimatedHeight` function used by the `src/domains/agents/components/node-view/dashboard/chartSubMenuIntersection.js` & `src/domains/rooms/components/room-overview/dashboard/chartSubMenuIntersection.js` in order to estimate the height. I don't know if it needs special handling additional to the `legend position/chartHeaderHeight` modifiers

Reply:
@apostolidhs I'm not sure either, but if it does - then perhaps you'll need (or we'll need) also to handle height from `localStorage.chart_height.{...}`? I think that handling localStorage will be much more common (here this is only about redirect from alarm notification). 

Also, in case you'll need that info - adding extra height based on `legendPosition` was for me just a temporary measure - in the future i suspect there will be need to also take dimensions.length into account. But i'm not 100% sure if it's possible at this time